### PR TITLE
Update main introduction and development intro sections.

### LINKED ISSUE #17: Work on the Introduction: What is Serverless? section
## Description

- Add content to the main Introduction section.
- Add content to the Introduction: What is Serverless? section.

**Filename:** ./source/introduction.md

**Page/Section:** 
- [x] Introduction
- [x] Introduction: What is Serverless?

**Details:** The content will cover:
- [x] Overview
- [] Misconceptions
- [] Characteristics
- [] Driving Factors/Benefits

### PR DESCRIPTION
**Pull request by rupakg**
## What did you implement:

- [x] Main Introduction section
- [x] Development intro section

Closes #17 and #18

## Todos:

- [x] Have you checked for typos and grammatical errors?
- [ ] If applicable, have you updated the TOC?

**Is this ready for review?:** Yes

### CODE REVIEW

File: source/dev/README.md [hunk 1]
```diff
@@ -1,12 +1,19 @@
 ## Development
 
-We as developers love to code. We hack on the newest cutting-edge technolgoies. We get excited about the future when we see a new technology like serverless. We jump at the 'hello-world' example, and are pumped when we get it going. Then we dream about using the technology to write our favorite application. That's when the painful path to developing a fully functioning application hits us. We pour over the documentation, read the blogs, and try some more examples. Yet it feels complicated and more like a wild goose chase. We lose sight of our application. We lose sight of our vision. We feel entangled in technology. We move on to the next one...
+Developing an application usually starts with an idea, which transforms into a code spike. A short sprint that manifests the core idea into code. Most developers want to share the idea, deploy the code as soon as they can. The excitement is usually curbed by the decisions that follow. The decisions around provisioning infrastructure, scalability, costs... That is usually a big chasm to cross.
```
Comment (worldsoup) [inline source/dev/README.md]: good idea with this start...really like that you're setting the stage for the core problem that serverless solves for developers (don't have to worry about infrastructure, faster time from idea to product). That being said, I think you can paint this picture more clearly...maybe try using a more specific example or more clearly articulating how this process unfolds and why it's frustrating for developers. At the end of this paragraph I should feel the pain

File: source/dev/README.md [hunk 2]
```diff
@@ -1,12 +1,19 @@
...
 
-**It does not have to be that way.** 
+The serverless computing paradigm is set to close that gap. It allows developers to think beyond infrastructure and keep their focus code-centric. It provides an architectural pattern, that allows writing code with functions being the unit of deployment. It allows the polygots in our midst, to have a choice to mix and match the code in different languages based on the needs of the application. With serverless, the code in a function can be written in any programming language that the serverless compute provider supports.
```
Comment (worldsoup) [inline source/dev/README.md]: again great idea but think the points can be made more clearly...take a look at this, Sudip does a really good job of articulating the serverless value https://medium.com/@chakrabartis/server-less-eats-the-infrastructure-2a3d0ef66363

File: source/dev/README.md [hunk 3]
```diff
@@ -1,12 +1,19 @@
...
 
-**It does not have to be that way.** 
+The serverless computing paradigm is set to close that gap. It allows developers to think beyond infrastructure and keep their focus code-centric. It provides an architectural pattern, that allows writing code with functions being the unit of deployment. It allows the polygots in our midst, to have a choice to mix and match the code in different languages based on the needs of the application. With serverless, the code in a function can be written in any programming language that the serverless compute provider supports. 
 
-The serverless movement has legs. We are poised to make the Serverless Framework the best in-class opensource offering. It makes developing & deploying applications fun again.
+The serverless architecture enables building applications one function at a time. The individual functions cater to a single responsibility. Functions related to each other by a common functional domain can be grouped together as a service. Albeit, the grouping, each function is still remains the unit of deployment.
```
Comment (worldsoup) [inline source/dev/README.md]: this is getting too framework specific...focus on function of the unit of deployment and why that matters/is valuable.

File: source/dev/README.md [hunk 4]
```diff
@@ -1,12 +1,19 @@
...
 
-The serverless movement has legs. We are poised to make the Serverless Framework the best in-class opensource offering. It makes developing & deploying applications fun again.
+The serverless architecture enables building applications one function at a time. The individual functions cater to a single responsibility. Functions related to each other by a common functional domain can be grouped together as a service. Albeit, the grouping, each function is still remains the unit of deployment.  
+
+The serverless style of computing is opening up the gates to a distributed, event-driven, collaborative ecosystem of functions and services. Serverless applications will interact with various event sinks and consume events from publishers. Developers will have access to a palette of services that they can discover, and integrate into their applications in real-time. Services made up of functions, possibly written in different programming languages, possibly running on different compute providers. All done in a transparent, seamless, cohesive manner.
```
Comment (worldsoup) [inline source/dev/README.md]: I like that you're laying out the future...I think you need to do a better job laying out the current state of serverless development first though.

**Comment by rupakg** [comment]
@nick thanks for your feedback. Will rewrite/add to it.

**Comment by rupakg** [comment]
@worldsoup can you also give me feedback on the main introduction piece: https://github.com/serverless/guide/pull/19/commits/88f5a138f498b57a10a6c90db1afbb92a474d7b4

**Pull request merged**